machine-runner/src/runner/runner.ts

```diff
@@ -662,6 +676,31 @@ namespace MachineRunnerIterableIterator {
       return Promise.resolve({ done: true, value: null })
     }
 
+    const actual = (): Promise<
+      IteratorResult<StateOpaque<SwarmProtocolName, MachineName, string, StateUnion>, null>
+    > => {
+      if (destruction.isDestroyed()) {
+        return Promise.resolve({ done: true, value: null })
+      }
+
+      if (
+        CommandGeneratorCriteria.allOk({
+          isCaughtUp: () => internals.caughtUp && internals.caughtUpFirstTime,
+          isQueueEmpty: () => internals.queue.length === 0,
+          isNotExpired: () => !ImplStateOpaque.isExpired(internals, internals.current),
+          isNotLocked: () => !ImplStateOpaque.isCommandLocked(internals),
+          isNotDestroyed: () => !ImplStateOpaque.isRunnerDestroyed(internals),
+        })
+      ) {
+        return Promise.resolve({
+          done: false,
+          value: ImplStateOpaque.make(internals, internals.current),
+        })
+      }
+
+      return nextValueAwaiter.peek()
```

Comment:
Could this be changed so that it also fires when a command emission expires? Basically, I’d want this to resolve as soon as commandAvailable returns true again.

Reply:
should be possible

Reply:
it is done. test is added.